include/pybind11/stl_bind.h

```diff
@@ -592,8 +603,26 @@ class_<Map, holder_type> bind_map(handle scope, const std::string &name, Args&&.
 
     cl.def(init<>());
 
+    cl.def(init([](const Map& d){
+            auto m = std::unique_ptr<Map>(new Map());
```

Comment:
Please use stack allocation and a normal ``Return`` at the end. This will be handled by C++'s return value optimization.